File: 3_1_1sort.cpp
//直接選択法p121

#include <stdio.h>
#include <iostream>

using namespace std;


#define N 8

int main(void){
    int a[]={80,40,35,12,88,1,56,33};
    int t;

     for(int i = 0; i < N-1; i++)
     {
        int min=a[i];
        int s=i;
        for(int j = i+1; j < N; j++)
        {
            if (a[j]<min) {
                min=a[j];
                s=j;
            }
            

        }
        t=a[i];
        a[i]=min;
        a[s]=t;
     }
     

     for(int i = 0; i < N; i++)
     {
         cout<<a[i]<<" ";
     }
     
    






}

File: 3_1_1sort.go
package main
import (
	"fmt"
)

func main(){
	a := [8]int{8,3,5,9,1,2,3,1}
	var t int

	for i:=0; i<8; i++{
		min := a[i]
		s := i
		for j:=i+1; j<8; j++{
			if a[j]<min{
				min=a[j]
				s=j
			}
		}

	t = a[i]
	a[i]=min
	a[s]=t
	}

	i:=0
	for i<8 {
		fmt.Println(a[i])
		i++
	}
}
